src/main/java/io/cdap/plugin/kudu/KuduSink.java

```diff
@@ -94,18 +93,13 @@ public KuduSink(KuduSinkConfig kuduSinkConfig) {
   @Override
   public void configurePipeline(PipelineConfigurer configurer) {
     super.configurePipeline(configurer);
-    Preconditions.checkArgument(!Strings.isNullOrEmpty(kuduSinkConfig.optSchema), "Write schema is not specified. Please add" +
-      "the write schema.");
+    FailureCollector failureCollector = configurer.getStageConfigurer().getFailureCollector();
+    kuduSinkConfig.validate(failureCollector);
 
     // Checks if that we are writing with has been constructed correctly.
     Schema writeSchema = kuduSinkConfig.getSchema();
     configurer.getStageConfigurer().setOutputSchema(writeSchema);
 
-    // If there is macro specified for 'master' address or table name, then
-    // we defer the creation of table to initialize.
-    if (kuduSinkConfig.containsMacro("master") || kuduSinkConfig.containsMacro("name")) {
```

Comment:
Why did we remove this again? Won't `createKuduTable` fail when those configs contain macro?

Reply:
I rollback this change